service/dap/server.go

```diff
@@ -623,10 +625,57 @@ func (s *Server) onPauseRequest(request *dap.PauseRequest) { // TODO V0
 	s.sendNotYetImplementedErrorResponse(request.Request)
 }
 
-// onStackTraceRequest sends a not-yet-implemented error response.
+// stackFrame represents the index of a frame within
+// the context of a stack of a specific goroutine.
+type stackFrame struct {
+	goroutineID int
+	frameIndex  int
+}
+
+// onStackTraceRequest handles ‘stackTrace’ requests.
 // This is a mandatory request to support.
-func (s *Server) onStackTraceRequest(request *dap.StackTraceRequest) { // TODO V0
-	s.sendNotYetImplementedErrorResponse(request.Request)
+func (s *Server) onStackTraceRequest(request *dap.StackTraceRequest) {
+	// TODO(polina): support depth and cfg via launch/attach args
+
+	goroutineID := request.Arguments.ThreadId
+	// Borrow the defaults for the arguments from the original vscode-go adapter.
+	// TODO(polina): is 'startFrame'+'levels' a better alternative for depth?
+	depth := 50 // length of the returned list of Stackframes
+	// TODO(polina): this is hardcoded throughout the codebase - make this a constant?
+	cfg := &api.LoadConfig{FollowPointers: true, MaxVariableRecurse: 1, MaxStringLen: 64, MaxArrayValues: 64, MaxStructFields: -1}
+	opts := api.StacktraceOptions(0)
+
+	locs, err := s.debugger.Stacktrace(goroutineID, depth, opts, api.LoadConfigToProc(cfg))
+	if err != nil {
+		s.sendErrorResponse(request.Request, UnableToProduceStackTrace, "Unable to produce stack trace", err.Error())
+		return
+	}
+
+	stackFrames := make([]dap.StackFrame, len(locs))
+	for i, loc := range locs {
+		uniqueStackFrameID := s.stackFrameHandles.create(stackFrame{goroutineID, i})
+		stackFrames[i] = dap.StackFrame{Id: uniqueStackFrameID, Line: loc.Line}
+		if loc.Function != nil {
+			stackFrames[i].Name = loc.Function.Name()
```

Comment:
Unless there's a specific reason why unknown functions should be called `<unknown>` you should call `loc.Function.Name()` directly, without the if.

Reply:
Done